### PR TITLE
Stripe Basic Setup

### PR DESCRIPTION
**Pull request by HAZberi**
### What this PR does (required):
- Initiate Stripe backend
- Create Checkout Session controller
- Create Customer controller 

### Screenshots / Videos (required):

https://user-images.githubusercontent.com/72773994/127888776-795d8e48-62e8-472e-bcc5-5b658d3bd721.mov

### Any information needed to test this feature (required):
- Use postman or curl to test the api routes for checkout session and customer creation

### Any issues with the current functionality (optional):
- Some refactoring will be required when integrating existing payment controllers with client. 
- This PR focuses on basic stripe setup only. Further modification to the backend will be carried out in stripe integration branch.

### CODE REVIEW

File: server/controllers/payment.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+});
+
+//create a customer
+exports.createCustomer = asyncHandler(async (req, res) => {
+  const userID = req.user.id;
+
+  try {
+    const user = await User.findById(userID);
+
+    const customer = await stripe.customers.create({
```
Comment (rajivtitus) [inline server/controllers/payment.js:48]: Perform a check to create a customer only if the logged in user does not already exist on stripe. Also, upon successful customer creation, you receive a stripe customer object, maybe update the User document with the stripe customer id?

**Review by rajivtitus — COMMENTED** [review]
Looks good for a start.

One comment, please have a look

**Review by rajivtitus — APPROVED** [review]
As discussed on our call, the changes will be addressed in a new PR. 
Approving this PR for now

File: server/controllers/payment.js (hunk 1 above)
Comment (HAZberi) [reply server/controllers/payment.js:48]: I will implement these changes in the next PR. Or alternatively pls checkout to stripe-integration.

**Pull request merged**